### PR TITLE
[melodic-devel] MoveItCpp backport

### PR DESCRIPTION
**Pull request by tylerjw**
### Description

This is a backport of MoveItCpp for melodic.  After talking with @henningkayser I realized we also need this released.  As far as I can tell backporting this does not break any existing API as it just extends a new API to a simplified internal interface.

This was a much simpler PR to build than the jog_arm one.

### CODE REVIEW

**Comment by v4hn** [comment]
@henningkayser Please review (and merge).

File: .travis.yml [hunk 1]
```diff
@@ -31,7 +32,13 @@ jobs:
       env: TEST=clang-tidy-fix
            BEFORE_DOCKER_SCRIPT="source moveit_kinematics/test/test_ikfast_plugins.sh"
            CXXFLAGS="-Wall -Wextra -Wwrite-strings -Wunreachable-code -Wpointer-arith -Wredundant-decls -Wno-unused-parameter -Wno-unused-function -Wno-overloaded-virtual"
-#           ROS_REPO=ros-shadow-fixed
+    - if: branch =~ /^(.*-devel|master)$/
```
Comment (henningkayser) [inline .travis.yml:35]: Should this really be part of this PR?

Comment (tylerjw) [reply .travis.yml:35]: These changes are also in my bug fixes PR and I cherry-picked them here to unify travis runs on melodic-devel.  Whichever one gets merged first I'll drop them out of the other PR.  

Would you rather wait on the other change to be merged and then I'll drop them out of here as they are unrelated to this change?

**Comment by tylerjw** [comment]
Pushed a rebase after the last merge.

**Pull request merged**